### PR TITLE
Shenli/refactor distinct

### PR DESCRIPTION
**Pull request by shenli**
@coocood @qiuyesuifeng

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

File: util/distinct/distinct_test.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package distinct
+
+import (
+	"testing"
+
+	"github.com/pingcap/check"
```
Comment (qiuyesuifeng) [inline util/distinct/distinct_test.go]: Why not using `. "github.com/pingcap/check"`

Comment (shenli) [inline util/distinct/distinct_test.go]: There is a Cheker in check package.

File: ast/functions.go [hunk 2]
```diff
@@ -542,69 +542,10 @@ func (a *AggregateFuncExtractor) Leave(n Node) (node Node, ok bool) {
...
-	a.Distinct, _ = memkv.CreateTemp(true)
-}
-
 // AggEvaluateContext is used to store intermediate result when caculation aggregate functions.
 type AggEvaluateContext struct {
-	Distinct  *AggregateDistinct
-	Count     int64
-	Value     interface{}
-	evaluated bool
+	distinctChecker *distinct.Checker
```
Comment (qiuyesuifeng) [inline ast/functions.go]: Seems no function to clear distinctChecker inner map member?

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**